### DISCUSSION TITLE
Expose Babel config for @astrojs/react, other enhancements to work with CSS-in-JS libraries

### DISCUSSION THREAD

**Discussion by verystack** [withastro/roadmap#202] [Proposal]
Currently it's *near* impossible to work with any CSS-in-JS library. Big and little (from Emotion, Styled Comps, to my favourite Compiled CSS-in-JS) - they all require Babel plugins and some level of control over the SSR renderer to allow for proper rendering of styles and critical css extraction.

This sort of functionality I believe isn't a nice-to-have - it's an essential - for any org with large existing component libraries for where it's impractical to use Tailwind/CSS/CSS Modules.

**Comment by bingalls** [comment]
It took me some time, to figure out why Chakra-ui lost all of its styling, when working with Astro.build.
Chakra depends on Emotion. It also gets positive reviews from Josh Comeau, a luminary in the CSS world.
Could an Astro plugin alternately solve this?

**Reply by verystack** [reply]
Not unless the plugin itself includes a renderer. Emotion requires extra steps in SSR rendering (as does any CSS-in-JS that can be SSR'd). Have a working fork of the renderer that includes this - but lacks babel yet for support of CSS prop.

**Comment by HummingMind** [comment]
+1 for this. Ran into this issue with Mantine UI and MUI (both React UI component libraries using emotion).
@natemoo-re confirmed in Discord they are aware of this issue and will look into addressing it after 1.0 launches.

**Comment by natemoo-re** [comment]
Thanks for the ping @HummingMind! Yes this is a pretty critical feature that we're missing. Obviously there are workarounds (don't use CSS-in-JS), but that's not an option for large portions of the ecosystem that have already bought into this pattern.

This is definitely something we would like to fix, but it unfortunately won't make the cut for v1.0. I expect this to be a very high priority for us as soon as v1.0 is out the door (currently planned for the end of July 2022).

**Reply by YowaiCoder** [reply]
Ping.
